code/modules/robotics/robot/robot_docking_station.dm

```diff
@@ -259,19 +278,31 @@ TYPEINFO(/obj/machinery/recharge_station)
 		if (src.occupant)
 			mainframe.deploy_to_shell(src.occupant)
 
-/obj/machinery/recharge_station/proc/build_icon()
+/obj/machinery/recharge_station/set_broken(mob/user)
+	. = ..()
+	if(.) return
+	AddComponent(/datum/component/equipment_fault/shorted, tool_flags = TOOL_SCREWING | TOOL_WIRING | TOOL_SNIPPING | TOOL_PRYING)
+	src.visible_message(SPAN_ALERT("[src] shutters and breaks down!"))
+	playsound(src, pick('sound/machines/glitch1.ogg','sound/machines/glitch2.ogg','sound/machines/glitch3.ogg'), 50, 2)
```

Comment:
not for now but this should probably be a proc on machinery/

Reply:
which changes should I consider for a new proc later on? The FX?

Reply:
yeah hence the comment on that line